packages/esm-service-queues-app/src/queue-table/cells/columns.resource.ts

```diff
@@ -0,0 +1,93 @@
+import { type TFunction, useTranslation } from 'react-i18next';
+import { type QueueTableColumn } from '../../types';
+import { queueTableNameColumn } from './queue-table-name-cell.component';
+import { queueTablePatientIdColumn } from './queue-table-patient-id-cell.component';
+import { queueTablePriorityColumn } from './queue-table-priority-cell.component';
+import { queueTableStatusColumn } from './queue-table-status-cell.component';
+import { queueTableComingFromColumn } from './queue-table-coming-from-cell.component';
+import { queueTableQueueColumn } from './queue-table-queue-cell.component';
+import { queueTableWaitTimeColumn } from './queue-table-wait-time-cell.component';
+import { useMemo } from 'react';
+import { type ColumnDefinition, type ConfigObject } from '../../config-schema';
+import { useConfig } from '@openmrs/esm-framework';
+import { activeVisitActionsColumn } from '../../active-visits/active-visits-row-actions.component';
+
+// returns the columns to display for a queue table of a particular queue + status.
+// For a table displaying all entries of a particular queue, the status param should be null
+// For a table displaying all entries from all queues, both params should be null
+export function useColumns(queue: string, status: string): QueueTableColumn[] {
+  const { t } = useTranslation();
+  const config = useConfig<ConfigObject>();
+  const { tablesConfig } = config;
+  const { columnDefinitions, tableDefinitions } = tablesConfig;
+
+  const columnsMap = useMemo(() => {
+    const map = new Map<string, QueueTableColumn>();
+    for (const columnDef of columnDefinitions) {
+      map.set(columnDef.id, getColumnFromDefinition(t, columnDef));
+    }
+    return map;
+  }, [columnDefinitions, t]);
```

Comment:
Do we need to get the `t` function here and pass it down, rather than just get it in the getColumnFromDefinition method directly?